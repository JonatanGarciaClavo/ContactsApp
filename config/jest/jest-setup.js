/*  eslint-disable */
// Make Enzyme functions available in all test files without importing
import { shallow, render, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
global.shallow = shallow;
global.render = render;
global.render = mount;
global.shallowToJson = shallowToJson;

// Skip createElement warnings but fail tests on any other warning
console.error = message => {
  if (!/(React.createElement: type should not be null)/.test(message)) {
      throw new Error(message);
  }
};