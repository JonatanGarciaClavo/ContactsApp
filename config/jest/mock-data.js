export const contact = {
  id: '-1',
  email: 'test@test.com',
  imgUrl: 'http://www.test.image.com',
  name: 'Tester',
  phoneNumber: '666666666',
};

export const contacts = [
  {
    id: '-1',
    email: 'test@test.com',
    imgUrl: 'http://www.test.image.com',
    name: 'Tester',
    phoneNumber: '666666666',
  },
  {
    id: '-2',
    email: 'test2@test.com',
    imgUrl: 'http://www.test2.image.com',
    name: 'Tester2',
    phoneNumber: '766666666',
  },
];

export const id = '-1';
