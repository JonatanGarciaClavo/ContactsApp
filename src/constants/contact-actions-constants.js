export const LOADING_CONTACT = 'LOADING_CONTACT';
export const INITIALIZE_CONTACT = 'INITIALIZE_CONTACT';
export const UPDATE_CONTACT_ATTRIBUTE = 'UPDATE_CONTACT_ATTRIBUTE';
export const VALIDATE_CONTACT_ATTRIBUTE = 'VALIDATE_CONTACT_ATTRIBUTE';
export const VALIDATE_CONTACT = 'VALIDATE_CONTACT';
export const REQUEST_CONTACT = 'REQUEST_CONTACT';
export const REQUEST_SAVE_CONTACT = 'REQUEST_SAVE_CONTACT';
