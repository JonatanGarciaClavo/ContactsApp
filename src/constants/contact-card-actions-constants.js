export const RESET_CONTACT_CARD = 'RESET_CONTACT_CARD';
export const REQUEST_CONTACT_CARD_SUCCESS = 'REQUEST_CONTACT_CARD_SUCCESS';
export const REQUEST_CONTACT_CARD = 'REQUEST_CONTACT_CARD';
export const REQUEST_DELETE_CONTACT_CARD = 'REQUEST_DELETE_CONTACT_CARD';
export const TRANSTION_TO_EDIT_CONTACT_CARD = 'TRANSTION_TO_EDIT_CONTACT_CARD';
