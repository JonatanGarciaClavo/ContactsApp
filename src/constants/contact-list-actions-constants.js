export const LOADING_CONTACT_LIST = 'LOADING_CONTACT_LIST';
export const INITIALIZE_CONTACT_LIST = 'INITIALIZE_CONTACT_LIST';
export const CHANGE_LIST_MODE = 'CHANGE_LIST_MODE';
