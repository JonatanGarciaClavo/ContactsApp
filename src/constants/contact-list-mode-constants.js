export const LIST_MODE = 'list';
export const CARD_MODE = 'card'
