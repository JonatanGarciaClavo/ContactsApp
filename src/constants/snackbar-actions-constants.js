export const CLOSE_ERROR_MESSAGE = 'CLOSE_ERROR_MESSAGE';
export const SET_ERROR_MESSAGE = 'SET_ERROR_MESSAGE';
