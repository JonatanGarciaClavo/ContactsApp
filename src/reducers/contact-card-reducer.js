import Immutable from 'immutable';
import { REQUEST_CONTACT_CARD, REQUEST_DELETE_CONTACT_CARD, REQUEST_CONTACT_CARD_SUCCESS,
  RESET_CONTACT_CARD } from '../constants/contact-card-actions-constants';
import { SET_ERROR_MESSAGE } from '../constants/snackbar-actions-constants';

export const initialState = new Immutable.Map({
  loading: false,
  contact: new Immutable.Map(),
});

const contactCard = (state = initialState, action) => {
  switch (action.type) {
    case (REQUEST_CONTACT_CARD):
    case (REQUEST_DELETE_CONTACT_CARD):
      return state.set('loading', true);
    case (SET_ERROR_MESSAGE):
      return state.set('loading', false);
    case (REQUEST_CONTACT_CARD_SUCCESS):
      return state
        .set('contact', action.contact)
        .set('loading', false);
    case (RESET_CONTACT_CARD):
      return initialState;
    default:
      return state;
  }
};

export default contactCard;
