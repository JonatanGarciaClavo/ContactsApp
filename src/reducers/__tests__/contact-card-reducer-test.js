import Immutable from 'immutable';
import { initialState, default as contactCard } from '../contact-card-reducer';
import { REQUEST_CONTACT_CARD, REQUEST_DELETE_CONTACT_CARD, REQUEST_CONTACT_CARD_SUCCESS,
  RESET_CONTACT_CARD } from '../../constants/contact-card-actions-constants';
import { SET_ERROR_MESSAGE } from '../../constants/snackbar-actions-constants';
import { contact } from '../../../config/jest/mock-data';

const testContact = new Immutable.Map(Immutable.fromJS(contact));

describe('Test contact card reducer', () => {
  it('returns the initialState state on an undefined state', () => {
    expect(contactCard(undefined, { type: '_NULL' })).toMatchSnapshot();
  });
  it('returns the same state on an unhandled action', () => {
    expect(contactCard(initialState, { type: '_NULL' })).toMatchSnapshot();
  });

  it('handles REQUEST_CONTACT_CARD action', () => {
    expect(contactCard(initialState, { type: REQUEST_CONTACT_CARD })).toMatchSnapshot();
  });
  it('handles REQUEST_DELETE_CONTACT_CARD action', () => {
    expect(contactCard(initialState, { type: REQUEST_DELETE_CONTACT_CARD })).toMatchSnapshot();
  });
  it('handles SET_ERROR_MESSAGE action', () => {
    expect(contactCard(initialState, { type: SET_ERROR_MESSAGE })).toMatchSnapshot();
  });

  it('handles REQUEST_CONTACT_CARD_SUCCESS action', () => {
    expect(contactCard(initialState, { type: REQUEST_CONTACT_CARD_SUCCESS, contact: testContact }))
      .toMatchSnapshot();
  });
  it('handles RESET_CONTACT_CARD action', () => {
    expect(contactCard(initialState, { type: RESET_CONTACT_CARD }))
      .toMatchSnapshot();
  });
});
