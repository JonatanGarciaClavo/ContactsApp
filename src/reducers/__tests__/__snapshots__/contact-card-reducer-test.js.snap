exports[`Test contact card reducer handles REQUEST_CONTACT_CARD action 1`] = `
Object {
  "contact": Object {},
  "loading": true,
}
`;

exports[`Test contact card reducer handles REQUEST_CONTACT_CARD_SUCCESS action 1`] = `
Object {
  "contact": Object {
    "email": "test@test.com",
    "id": "-1",
    "imgUrl": "http://www.test.image.com",
    "name": "Tester",
    "phoneNumber": "666666666",
  },
  "loading": false,
}
`;

exports[`Test contact card reducer handles REQUEST_DELETE_CONTACT_CARD action 1`] = `
Object {
  "contact": Object {},
  "loading": true,
}
`;

exports[`Test contact card reducer handles RESET_CONTACT_CARD action 1`] = `
Object {
  "contact": Object {},
  "loading": false,
}
`;

exports[`Test contact card reducer handles SET_ERROR_MESSAGE action 1`] = `
Object {
  "contact": Object {},
  "loading": false,
}
`;

exports[`Test contact card reducer returns the initialState state on an undefined state 1`] = `
Object {
  "contact": Object {},
  "loading": false,
}
`;

exports[`Test contact card reducer returns the same state on an unhandled action 1`] = `
Object {
  "contact": Object {},
  "loading": false,
}
`;
